src/Compilers/Core/Portable/SourceGeneration/AdditionalSourcesCollection.cs

```diff
@@ -15,19 +15,16 @@ internal sealed class AdditionalSourcesCollection
     {
         private readonly ArrayBuilder<GeneratedSourceText> _sourcesAdded;
 
+        private readonly string _fileExtension;
+
         private const StringComparison _hintNameComparison = StringComparison.OrdinalIgnoreCase;
 
         private static readonly StringComparer s_hintNameComparer = StringComparer.OrdinalIgnoreCase;
 
-        internal AdditionalSourcesCollection()
+        internal AdditionalSourcesCollection(string fileExtension)
         {
```

Comment:
When dealing with extensions can never remember if we should or should not include the `.`. So `".cs"` vs. `"cs"`. Consider either enforcing one or the other is passed here or normalizing the input to a specific one.